src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,454 @@
+ * 'even-segments'- colored tracks shown in segments 2,4,6, etc. Default if number of Thumbs values are even.
+ */
+export type TrackDisplayMode =
+  | 'auto'
+  | 'none'
+  | 'odd-segments'
+  | 'even-segments';
+
+/** Private Utility Functions */
+const getPercentageOfRectangle = (rect: DOMRect, pointer: number) => {
+  const position = getBoundedValue(pointer, rect.left, rect.right);
+  return (position - rect.left) / rect.width;
+};
+
+const getClosestValueIndex = (values: number[], pointerValue: number) => {
+  if (1 === values.length) {
+    return 0;
+  }
+  const distances = values.map((value) => Math.abs(value - pointerValue));
+  const smallest = Math.min(...distances);
+  return distances.indexOf(smallest);
+};
+
+const getDefaultTrackDisplay = (
+  trackDisplayMode?: TrackDisplayMode,
+  values?: number[],
+) => {
+  const numValues = values?.length ?? 0;
+  if (undefined !== trackDisplayMode && 'auto' !== trackDisplayMode) {
+    return trackDisplayMode;
+  }
+
+  return numValues % 2 ? 'even-segments' : 'odd-segments';
+};
+
+const roundValueToClosestStep = (value: number, step: number, min: number) => {
+  return Math.round((value - min) / step) * step + min;
+};
+
+const formatNumberValue = (
+  value: number,
+  step: number,
+  numDecimals: number,
+) => {
+  if (Number.isInteger(step)) {
+    return value.toFixed(0);
+  }
+  return value.toFixed(numDecimals);
+};
+
+/**
+ * Properties for Slider component
+ */
+export type SliderProps = {
+  /**
+   * Set focus on first thumb in slider element.
+   * @default false
+   */
+  setFocus?: boolean;
+  /**
+   * Minimum slider value.
+   * @default 0
+   */
+  min?: number;
+  /**
+   * Maximum slider value.
+   * @default 100
+   */
+  max?: number;
+  /**
+   * Array of one or more values to show.
+   */
+  values: number[];
+  /**
+   *  Determines which segments are shown with color.
+   * 'none' - no colored tracks are displayed.
+   * 'auto' - no colored tracks are displayed.
+   * 'odd-segments'- colored tracks shown in segments 1,3,5, etc.
+   *                 Default if number of thumbs values are even.
+   * 'even-segments'- colored tracks shown in segments 0,2,4, etc.
+   *                 Default if number of thumbs values are odd.
+   * @default 'auto'
+   */
+  trackDisplayMode?: TrackDisplayMode;
+  /**
+   * Step increment controls what values are allowed and the amount the value will
+   * change when left and right arrows are pressed when a Thumb has focus.
+   * @default 1
+   */
+  step?: number;
+  /**
+   * Forces control to be displayed in a disabled state where no interactive value
+   * changes are allowed.
+   *  @default false
+   */
+  disabled?: boolean;
+  /**
+   * Props to override default for showing a tooltip when Thumb is active or has focus.
+   * changes are allowed.
+   */
+  tooltipProps?: Partial<Omit<TooltipProps, 'content' | 'children'>>;
+  /**
+   * Optional tooltip function that can be supplied to generate tooltip text.
```

Comment:
Not sure I like that Optional mentioning in some props.. Some has, some not.. Maybe just remove that, as the prop definition shows if its optional...